File: code/SlidingWindow/MaximumSumSubarray/MaximumSumSubarray.js
const maxSubArrayOfSizeK = function(k, array){
    let maxSum = 0;
    let currentSum = 0;
    let startingPoint = 0;

    for(let endingPoint = 0; endingPoint < array.length; endingPoint++){
        currentSum += array[endingPoint];
        if(endingPoint >= k - 1){
            maxSum = Math.max(maxSum, currentSum);
            currentSum -= array[startingPoint];
            startingPoint ++;
        }
        return maxSum;
    }
}

File: code/SlidingWindow/MaximumSumSubarray/MaximumSumSubarray.java
package code.SlidingWindow.MaximumSumSubarray;

public class MaximumSumSubarray {
    public static int findMaxSumArrayOfSizeK(int k, int[] array){
        int maxSum = 0;
        int currentSum = 0;
        int startingPoint = 0;

        for(int endingPoint = 0; endingPoint < array.length; endingPoint ++){
            currentSum += array[endingPoint];
            if(endingPoint >= k - 1){
               maxSum = Math.max(currentSum, maxSum);
               currentSum -= array[startingPoint];
               startingPoint++; 
            }
        }
        return maxSum;
    }
}